lib/experimental/fonts-library/class-wp-fonts-library.php

```diff
@@ -0,0 +1,78 @@
+<?php
+/**
+ * Fonts Library class.
+ *
+ * This file contains the Fonts Library class definition.
+ *
+ * @package    WordPress
+ * @subpackage Fonts Library
+ * @since      6.4.0
+ */
+
+if ( class_exists( 'WP_Fonts_Library' ) ) {
+	return;
+}
+
+/**
+ * Fonts Library class.
+ *
+ * @since 6.4.0
+ */
+class WP_Fonts_Library {
+
+	const ALLOWED_FONT_MIME_TYPES = array(
+		'otf'   => 'font/otf',
+		'ttf'   => 'font/ttf',
+		'woff'  => 'font/woff',
+		'woff2' => 'font/woff2',
+	);
+
+	/**
+	 * Gets the upload directory for fonts.
+	 *
+	 * @since 6.4.0
+	 *
+	 * @return string Path of the upload directory for fonts.
+	 */
+	public static function get_fonts_dir() {
+		return wp_upload_dir()['basedir'] . '/fonts';
+	}
+
+	/**
+	 * Sets the upload directory for fonts.
+	 *
+	 * @since 6.4.0
+	 *
+	 * @param array $defaults {
+	 *     Default upload directory.
+	 *
+	 *     @type string $path    Path to the directory.
+	 *     @type string $url     URL for the directory.
+	 *     @type string $subdir  Sub-directory of the directory.
+	 *     @type string $basedir Base directory.
+	 *     @type string $baseurl Base URL.
+	 * }
+	 * @return array Modified upload directory.
+	 */
+	public static function set_upload_dir( $defaults ) {
+		$defaults['subdir'] = '/fonts';
+		$defaults['path']   = $defaults['basedir'] . $defaults['subdir'];
+		$defaults['url']    = $defaults['baseurl'] . $defaults['subdir'];
+
+		return $defaults;
+	}
+
+	/**
+	 * Registers the fonts library post type.
+	 *
+	 * @since 6.4.0
+	 */
+	public static function register_post_type() {
+		$args = array(
+			'public'       => true,
+			'label'        => 'Font Library',
+			'show_in_rest' => true,
+		);
+		register_post_type( 'wp_font_family', $args );
+	}
+}
```

Comment:
Core registers all post types in its `create_initial_post_types()` function. And the code for registration is (and should be) contained within that function, rather than a call to another function or method. Thus, this static method can be removed and its code relocated to the init function.

Commit is coming shortly.

Reply:
Done ✅ https://github.com/WordPress/gutenberg/pull/52704/commits/54535792a27a758e2633caf21ce632ae7cf94e86

Reply:
I fear that this code may unintentionally expose the endpoint broadly. For example, issues like #54526 are occurring.

I think it would be better if `public` is `false` and `_builtin` is `true` ?

Reply:
Good catch thanks @t-hamano ! Submitted a PR for this: https://github.com/WordPress/gutenberg/pull/54559